src/package.rs

```diff
@@ -329,32 +95,46 @@ impl PartialOrd for Package {
     }
 }
 
-impl PartialEq for Package {
-    fn eq(&self, other: &Self) -> bool {
-        self.full_name() == other.full_name() && self.source == other.source
-    }
-}
-
-impl Eq for Package {}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Packages(pub Vec<Package>);
-
-impl From<Vec<Package>> for Packages {
-    fn from(x: Vec<Package>) -> Self {
-        Packages(x)
-    }
-}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Packages(Vec<Package>);
 
 impl Packages {
-    pub fn iter(&self) -> Iter<Package> {
-        self.0.iter()
-    }
+    /// An empty list.
+    pub fn empty() -> Self {
+        Self(vec![])
+    }
+
+    /// Parse a line into a list of packages.
+    pub fn from_line(line: &str) -> Self {
+        // First we split the line into separating characters
+        let lines = line
+            .split(|c| c == ';' || c == '|' || c == '&')
+            // Then try to find the proper package manager for each line, this also filters out
```

Comment:
Nice solution! And much faster, then regexes. But it can be a lot of tricky moments, a-la `bash -c 'cargo install package'`. So maybe we need to use something like [this](https://github.com/ipetkov/conch-parser). Or it's an overkill (:

Reply:
I was considering this as well but I also want to support different shells..

Reply:
I was considering this as well but I also want to support different shells..